docs/content/api-overview/resources/nsg.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Network Security Group"
+date: 2020-07-19T16:44:00-04:00
+weight: 5
+chapter: false
+---
+
+#### Overview
+The Network Security Group builder creates network security groups with rules for securing network access to resources.
+
+* Network Security Groups (`Microsoft.Network/networkSecurityGroups`)
+* Security Rules (`Microsoft.Network/networkSecurityGroups/securityRules`)
+
+#### Builder Keywords
+
+| Applies To | Keyword | Purpose |
+|-|-|-|
+| nsg | name | Specifies the name of the network security group |
+| nsg | add_rules | Adds security rules to the network security group |
+| securityRule | name | The name of the security rule |
+| securityRule | description | The description  of the security rule |
+| securityRule | service | The service port(s) and protocol(s) protected by this security rule |
+| securityRule | add_source | Specify access from any source protocol, address, and port |
+| securityRule | add_source_any | Specify access from any address and any port |
+| securityRule | add_source_address | Specify access from a specific address and any port |
+| securityRule | add_source_network | Specify access from a specific network and any port |
+| securityRule | add_source_tag | Specify access from a tagged source such as "Internet", "VirtualNetwork", or "AzureLoadBalancer" |
+| securityRule | add_destination | Specify access to any source protocol, address, and port |
+| securityRule | add_destination_any | Specify access to any address and any port |
+| securityRule | add_destination_address | Specify access to a specific address and any port |
+| securityRule | add_destination_network | Specify access from a specific network and any port |
+| securityRule | add_destination_tag | Specify access to a tagged destination such as "Internet", "VirtualNetwork", or "AzureLoadBalancer" |
+| securityRule | allow | Allows this traffic (the default) |
+| securityRule | deny | Denies this traffic |
+
+#### Basic Example
+
+```fsharp
+open Farmer
+open Farmer.Builders
+open Farmer.NetworkSecurity
+
+// Create a rule for https services accessible from the internet
+let httpsAccess = securityRule {
+    name "web-servers"
+    service (Service ("https", Port 443us))
```

Comment:
Can probably add an overload here to simply take in string and port number?

Reply:
Also consider replacing "https" with a DU (HTTP | HTTPS) (if we don't already have one). Or could we can Service itself to being a union of HTTP / HTTPS? Or, have two keywords, `http_service` and `https_service`.

Reply:
Ignore that comment - I can see that the HTTPS thing is just the name of the service! However the comment on the overload / tuple is still valid.